exchanges/gateio/gateio_wrapper.go

```diff
@@ -1008,11 +1012,20 @@ func (g *Gateio) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Submi
 		if err != nil {
 			return nil, err
 		}
+
+		// When doing spot market orders when purchasing base currency, the
+		// quote currency amount is used. When selling the base currency the
+		// base currency amount is used.
+		tradingAmount := s.Amount
+		if tradingAmount == 0 {
+			tradingAmount = s.QuoteAmount
+		}
```

Comment:
Just a mini one on this.
> When type is limit, it refers to base currency. For instance, BTC_USDT means BTC

https://www.gate.io/docs/developers/apiv4/#create-an-order

The requirements that you've added kinda allow an invalid limit order through. I would request that you add another requirement just to avoid a s.QuoteAmount limit order through that tries to purchase 1000BTC instead of 1000USD worth of BTC

I would also accept
```go
if tradingAmount == 0 && s.Type == order.Market {
```